integrated_channels/canvas/client.py

```diff
@@ -358,33 +365,34 @@ def _extract_integration_id(self, data):
 
         return integration_id
 
-    def _find_course_by_course_id(self, edx_course_id):
+    def _find_course_in_account(self, edx_course_id, canvas_account_id):
         """
-        Search course by edx_course_id (used as integration_id in canvas) under given account.
-          Issue with this is that if this course exists in canvas with this edx_course_id
-          under a different subaccount, we will not detect it.
-          Hence we will try to find a course under other subaccounts in case we don't find it
-          under the current subaccount
-          Will even return courses that are in the 'deleted' state in Canvas, so we can correctly
-          skip these courses in logic for create_or_update when needed.
+        Search course by edx_course_id (used as integration_id in canvas) under provided account.
+        It will even return courses that are in the 'deleted' state in Canvas, so we can correctly
+        skip these courses in logic as needed.
 
-          Note: we do not need to follow pagination here since it would be extremely unlikely
-          that searching by a specific edx_course_id results in many records, we generally only
-          expect 1 record to come back anyway.
+        Note: we do not need to follow pagination here since it would be extremely unlikely
+        that searching by a specific edx_course_id results in many records, we generally only
+        expect 1 record to come back anyway.
 
-          The `&state[]=all` is added so we can also fetch priorly 'delete'd courses
+        The `&state[]=all` is added so we can also fetch priorly 'delete'd courses as well
         """
         url = "{}/api/v1/accounts/{}/courses/?search_term={}&state[]=all".format(
             self.enterprise_configuration.canvas_base_url,
-            self.enterprise_configuration.canvas_account_id,
+            canvas_account_id,
             quote(edx_course_id),
         )
         resp = self.session.get(url)
         all_courses_response = resp.json()
 
         if resp.status_code >= 400:
+            message = ''
+            if 'reason' in all_courses_response:
```

Comment:
it seems that when we hit a 404 we don't get a reason back, we just get 'errors' as an array